.github/workflows/docker.yml

```diff
@@ -27,39 +27,43 @@ jobs:
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Config
+      id: docker-build-config
       env:
         JOB:  ${{ github.job }}
         REPO: ${{ github.repository }}
       run: |
         set -e
-        git submodule foreach --recursive -q pwd                    \
-        | cat <(pwd) -                                              \
-        | sed -n 's/\(\/cris\-core\)\/*$/\1/p'                      \
-        | xargs -I{} find {} -maxdepth 1 -name docker -type d       \
-        | sort -u                                                   \
-        | head -n1                                                  \
-        | xargs -I{} docker build                                   \
-            --build-arg BASE_IMAGE="$(cut -d- -f2- <<< "$JOB"       \
-                    | sed 's/\([0-9]\)/:\1/'                        \
-                    | sed 's/\([0-9][0-9]\)\([0-9]\)/\1\.\2/'       \
-                )"                                                  \
-            --no-cache                                              \
-            --pull                                                  \
-            --rm                                                    \
-            -t "ghcr.io/$REPO-build-$(cut -d- -f2- <<< "$JOB"):ci"  \
-            {}
-    - name: Login
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      if: ${{ github.event_name != 'pull_request' }}
-      run: docker login ghcr.io -u "$GITHUB_ACTOR" -p "$GITHUB_TOKEN"
-    - name: Push
-      env:
-        JOB:  ${{ github.job }}
-        REPO: ${{ github.repository }}
-      if: ${{ github.event_name != 'pull_request' }}
-      run: docker push "ghcr.io/$REPO-build-$(cut -d- -f2- <<< "$JOB"):ci"
+        git submodule foreach --recursive -q pwd                                \
+        | cat <(pwd) -                                                          \
+        | sed -n 's/\(\/cris\-core\)\/*$/\1/p'                                  \
+        | xargs -I{} find {} -maxdepth 1 -name docker -type d                   \
+        | sort -u                                                               \
+        | head -n1                                                              \
+        | xargs -I{} echo "::set-output name=DOCKER_FILE_PATH::{}/Dockerfile"
+        echo "::set-output name=TAG::$REPO-build-$(cut -d- -f2- <<< "$JOB")"
+        echo "::set-output name=BASE_IMAGE::$(cut -d- -f2- <<< "$JOB"         \
+                    | sed 's/\([0-9]\)/:\1/'                                    \
+                    | sed 's/\([0-9][0-9]\)\([0-9]\)/\1\.\2/'                   \
+             )"
+    - name: Build and Push
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        file: ${{ steps.docker-build-config.outputs.DOCKER_FILE_PATH }}
+        build-args: "BASE_IMAGE=${{ steps.docker-build-config.outputs.BASE_IMAGE }}"
+        platforms: linux/arm64
```

Comment:
I build debian11 arm64 only in [Docker run #76](https://github.com/cyfitech/cris-core/actions/runs/2784415648) while arm+amd64 in [Docker run #75](https://github.com/cyfitech/cris-core/actions/runs/2780044087). The build time does not change much so I think we can just build amd and arm images together instead of putting them into separate tasks.

Reply:
IIRC our docker build only takes 10+ min before, not 1+ hour

Reply:
https://github.com/cyfitech/cris-core/actions/runs/2771406450

Reply:
Yes, but that's amd64 only. Building ccache for arm64 on qemu seems to be extremely slow.

Reply:
I guess it's just whatever computationally heavy is not q to emu.

Reply:
(also could you use aarch64 instead of arm64, I really cannot distinguish it from amd64....)

Reply:
So you'll probably have to cross-compile somehow

Reply:
Figuring out cross compiling and so may be much more expensive than the 1.5h build time. Most of the time the Docker build  won't be triggered so it won't be a blocker for PRs.

Reply:
It'll be painful for whoever working on dockerfile next time. 1.5h build time is not acceptable. If you don't want to cross-compile, you may have to skip ccache building and use prebuilt one, but make sure you carefully handle the cache if used in CI or it may overflow our cache quota and slow everything down.

Reply:
If we don't have a good way around then we should simply don't add it into CI for now.

Reply:
Removing ccache build will reduce the aarch64 image build time from 1.5h to about 25 min. I guess that is acceptable now.

Reply:
Yes, make sure downstream CI job can fallback when requested ccache flags is not available.

Reply:
We have installed it in apt-get so at least they will have older version of ccache.